From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: security <security@uniontech.com>
Date: Wed, 19 Aug 2026 00:00:00 +0800
Subject: [PATCH] protect usec-ac-manager policy directories on usec path

Add dedicated usec policy for:
- /var/lib/uos/usec
- /usr/share/uos/usec
- /etc/uos/usec

---
 policy/modules/services/deepin_perm_control.fc | 6 ++++++
 policy/modules/services/deepin_perm_control.te | 10 ++++++++++
 2 files changed, 16 insertions(+)

Index: refpolicy/policy/modules/services/deepin_perm_control.fc
===================================================================
--- refpolicy.orig/policy/modules/services/deepin_perm_control.fc
+++ refpolicy/policy/modules/services/deepin_perm_control.fc
@@ -12,4 +12,10 @@ ifdef(`enable_usec',`
 /etc/apt/auth\.conf\.d/uos\.conf   --   gen_context(system_u:object_r:deepin_uos_auth_conf_t,s0)
 /var/uos/\.license\.xkey           --   gen_context(system_u:object_r:deepin_license_xkey_t,s0)
+/var/lib/uos/usec                    -d   gen_context(system_u:object_r:deepin_uos_usec_policy_t,s0)
+/var/lib/uos/usec(/.*)?                   gen_context(system_u:object_r:deepin_uos_usec_policy_t,s0)
+/usr/share/uos/usec                  -d   gen_context(system_u:object_r:deepin_uos_usec_policy_t,s0)
+/usr/share/uos/usec(/.*)?                 gen_context(system_u:object_r:deepin_uos_usec_policy_t,s0)
+/etc/uos/usec                        -d   gen_context(system_u:object_r:deepin_uos_usec_policy_t,s0)
+/etc/uos/usec(/.*)?                       gen_context(system_u:object_r:deepin_uos_usec_policy_t,s0)
 ')

Index: refpolicy/policy/modules/services/deepin_perm_control.te
===================================================================
--- refpolicy.orig/policy/modules/services/deepin_perm_control.te
+++ refpolicy/policy/modules/services/deepin_perm_control.te
@@ -994,3 +994,13 @@ ifdef(`enable_usec',`
     allow deepin_corefile_writable_domain deepin_uos_auth_conf_t:file { read_file_perms setattr_file_perms write_file_perms };
     allow deepin_corefile_writable_domain deepin_license_xkey_t:file { read_file_perms setattr_file_perms write_file_perms };
 ')
+
+ifdef(`enable_usec',`
+    type deepin_uos_usec_policy_t;
+    deepin_file_type_set(deepin_uos_usec_policy_t)
+    deepin_readable_file_type_set(deepin_uos_usec_policy_t)
+
+    allow { deepin_perm_manager_sidtwo_t deepin_immutable_t dpkg_t sysadm_t security_t } deepin_uos_usec_policy_t:dir manage_dir_perms;
+    allow { deepin_perm_manager_sidtwo_t deepin_immutable_t dpkg_t sysadm_t security_t } deepin_uos_usec_policy_t:file manage_file_perms;
+    allow { deepin_perm_manager_sidtwo_t deepin_immutable_t dpkg_t sysadm_t security_t } deepin_uos_usec_policy_t:lnk_file manage_lnk_file_perms;
+')
